solarforecastarbiter/datamodel.py

```diff
@@ -1253,26 +1318,41 @@ class ReportFigure(BaseModel):
     div: str
     svg: str
     figure_type: str
+    figure_class: str = 'bokeh'
     category: str = ''
     metric: str = ''
 
 
+def __bokeh_or_plotly__(cls):
+    if cls.bokeh_version is not None and cls.plotly_version is not None:
+        raise KeyError('Only provide one of "bokeh_version" or '
+                       '"plotly_version" to RawReportPlots')
+    elif cls.bokeh_version is None and cls.plotly_version is None:
+        raise KeyError('Must provide one of "bokeh_version" or '
+                       '"plotly_version" to RawReportPlots')
+
+
 @dataclass(frozen=True)
 class RawReportPlots(BaseModel):
     """Class for storing collection of all metric plots on a raw report.
 
     Parameters
     ----------
-    bokeh_version: str
-        The bokeh version used when generating the plots.
-    script: str
-        The html script tag containing all of the bokeh javascript for the
-        plots.
     figures: tuple of :py:class:`solarforecastarbiter.datamodel.ReportFigure`
+    plotly_version: str
+        The plotly version used when generating metrics plots.
```

Comment:
``bokeh_version``is still a parameter?

Reply:
I've set this up such that it supports `bokeh_version` and `script` or just plotly version. I deleted the `bokeh_datamodel.py` that I had left in. It seemed like we wanted to support continuing to use Bokeh for users who interact directly with the core library?
The reports submodules that process and template the report as well as the actual template files have been altered to only work with the plotly stuff, but if a user was so inclined to create their own Bokeh based report outside of the SFA platform they would have the tools to do so with some slight modifications.